play/src/front/Components/Video/PictureInPicture.svelte

```diff
@@ -0,0 +1,259 @@
+                pipWindowStore.set(pipWindow);
+
+                // Listen the event whent the user whant to close the picture in picture mode
+                pipWindow.addEventListener("pagehide", destroyPictureInPictureComponent);
+            })
+            .catch((error: Error) => {
+                console.info("Picture-in-Picture is not supported", error);
+                destroyPictureInPictureComponent();
+                // Maybe we could propose a popup to the user to activare the Picture in Picture mode
+            });
+    });
+
+    onDestroy(() => {
+        destroyPictureInPictureComponent();
+        if (pipWindowStoreSubscriber) pipWindowStoreSubscriber();
+    });
+</script>
+
+<div
+    bind:this={divElement}
+    class="tw-h-[84%] tw-w-full tw-flex tw-justify-center tw-items-center tw-z-40 tw-py-8"
+    style="display: none;"
+>
+    <div
+        id="wrapper"
+        class="tw-w-4/5 tw-h-full tw-justify-center tw-items-center tw-grid tw-content-center tw-grid-cols-1 sm:tw-grid-cols-2 md:w-grid-cols-3 xl:w-grid-cols-4 tw-gap-3"
+        class:!tw-grid-cols-1={$streamableCollectionStore.size === 1}
+    >
+        {#each [...$streamableCollectionStore] as [uuid, streamable] (uuid)}
+            <div
+                id={`main-${streamable.uniqueId}`}
+                class="tw-w-full tw-h-auto tw-aspect-video tw-relative tw-flex tw-justify-center tw-items-center tw-z-40 tw-rounded-xl tw-bg-[#373a3e]"
+            >
+                <img
+                    src={getPlayerWokaPicture(streamable)}
+                    class="tw-w-auto tw-h-full tw-max-h-[8rem]"
+                    alt="woka user"
+                />
+            </div>
+        {/each}
+    </div>
+    <div
+        id="mycamera-wrapper"
+        class="tw-absolute tw-h-auto tw-w-1/5 tw-justify-center tw-items-center tw-right-[5%] tw-bottom-[20%] tw-flex tw-bg-[#373a3e] tw-z-40 tw-rounded-xl tw-aspect-video"
+    >
+        <img src={getMyPlayerWokaPicture()} class="tw-w-auto tw-h-full tw-max-h-[4rem]" alt="woka user" />
+        <div class="voice-meter-my-container tw-justify-end tw-z-[251] tw-pr-2 tw-absolute tw-w-full">
+            {#if $mediaStreamConstraintsStore.audio}
+                <SoundMeterWidget volume={$localVolumeStore} classcss="tw-absolute" barColor="blue" />
+            {:else}
+                <img draggable="false" src={microphoneOffImg} class="tw-flex tw-p-1 tw-h-8 tw-w-8" alt="Mute" />
+            {/if}
+        </div>
+        {#if myLocalStream != undefined}
+            <video
+                bind:this={myLocalStreamElement}
+                id="my-video"
+                class="tw-h-full tw-w-full tw-rounded-lg tw-object-cover tw-absolute"
+                style="-webkit-transform: scaleX(-1);transform: scaleX(-1);"
+                use:srcObject={myLocalStream}
+                autoplay
+                muted
+                playsinline
+            />
+        {/if}
+    </div>
+    <ActionBar isPictureInPicture={true} />
```

Comment:
We can discuss that when we merge with the "refonte-main" branch, but I'm wondering if we shouldn't have 2 different action bars (the new action bar is made of many different components and is less massive)

Reply:
I don't know about this specific component, but I know that the goal is to reuse the video and action components from the core and have the same design. It feels like we've never left the WorklAdventure meeting.